app/src/main/java/ch/hikemate/app/utils/LocationUtils.kt

```diff
@@ -0,0 +1,220 @@
+package ch.hikemate.app.utils
+
+import android.content.Context
+import android.location.Location
+import android.os.Looper
+import android.util.Log
+import android.widget.Toast
+import ch.hikemate.app.R
+import ch.hikemate.app.ui.map.MapScreen
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.MultiplePermissionsState
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.Priority
+import com.google.android.gms.tasks.CancellationToken
+import com.google.android.gms.tasks.CancellationTokenSource
+import org.osmdroid.views.MapView
+import org.osmdroid.views.overlay.Marker
+
+object LocationUtils {
+  private const val LOG_TAG = "LocationUtils"
+
+  /**
+   * Util function to check if the user has granted the location permission.
+   *
+   * Does not check whether fine location permission has been granted, checks that at least coarse
+   * location was granted.
+   *
+   * @param locationPermissionState The state of the location permission
+   * @return True if the user has granted the location permission, false otherwise
+   */
+  @OptIn(ExperimentalPermissionsApi::class)
+  fun hasLocationPermission(locationPermissionState: MultiplePermissionsState): Boolean {
```

Comment:
Fair question, I think `LocationUtils.kt` is more suited, because it is all about the location permission. In some sense, it is a special case of the `PermissionsUtils.kt`.

Reply:
LGTM